### DISCUSSION TITLE
PW stops running when monitors are off, snapshots are not persistent.

### DISCUSSION THREAD

**Discussion by svtdoug** [kangyu-california/PersistentWindows#177] [General]
I appreciate you making this app available.  Its seems to do what I need, but not entirely.  PW does seem to save and restore screen snapshots of running applications.  But I have 2 problems.  First, is that after running PW and capturing snapshot, I turn off monitors, returning the next day to turn on monitors - windows have all migrated to monitor 1, but PW is no longer running.  No longer in sys tray.  Why did it stop?  Second problem is after restarting PW, my previously saved snapshots are gone and restore snapshot is grayed out.   Why are the snapshots not persistent from one session to the other?  Are they not saved to disk?  

Any clues would be appreciated.

Doug

**Comment by kangyu-california** [comment]
1. If PW is invoked through task scheduler, it may get killed by OS due to not turning off "Stop the task if it runs longer than 3 days" in task setting
![image](https://user-images.githubusercontent.com/59128756/188043145-d2954a75-821d-4464-b07f-2001d1a3053b.png)
Or if PW really crashed, you may look up the application log in Event Viewer and search for .NET Framework error message which contains call stack information

2. As mentioned in readme, snapshot is saved in memory instead of on hard drive, i.e. snapshot is non-persist. Snapshots contain extra window z-order information which will no longer apply after computer reboot. For persistence storage, user need to capture window to disk. PW offers capability to capture multiple "named" window layout per display configuration, checkout online help for more details.

**Comment by svtdoug** [comment]
Thank you [kangyu-california](https://github.com/kangyu-california). I had not noticed the Capture Window to Disk option.  My bad.  That takes care of that issue.  

Also, I do not have the Stop the Task if it runs longer...  checked.  
 
As for crashing - here are the 9990 and 9999 events which occurred just before the crash:

9999 at 12:20:03
The description for Event ID 9999 from source Application cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.
If the event originated on another computer, the display information had to be saved with the event.
The following information was included with the event: 
PersistentWindows: no qualified position data to restore minimized window "Display Reception Reports — Mozilla Firefox"
The message resource is present but the message was not found in the message table

9990 at 12:23:13
The description for Event ID 9990 from source Application cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.
If the event originated on another computer, the display information had to be saved with the event.
The following information was included with the event: 
PersistentWindows: Start restoring window layout back to 9/5/2022 12:19:13 for display setting Display_Loc0x0_Res1920x1080__Display_Loc1920xM1080_Res4096x2160__Display_Loc6016x0_Res1920x1080
The message resource is present but the message was not found in the message table

9990 at 12:23:13
The description for Event ID 9990 from source Application cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.
If the event originated on another computer, the display information had to be saved with the event.
The following information was included with the event: 
PersistentWindows: Restore finished in pass 2 with 14 windows recovered for display setting Display_Loc0x0_Res1920x1080__Display_Loc1920xM1080_Res4096x2160__Display_Loc6016x0_Res1920x1080
The message resource is present but the message was not found in the message table

Thanks much! 

Doug

**Comment by kangyu-california** [comment]
1. Make sure you are running the latest 5.38 release
2. If crash can be reproduced, please search for events with crash info like following
![image](https://user-images.githubusercontent.com/59128756/188511986-24a497d2-1d2e-4d1e-a306-701b612233d9.png)
Or
![image](https://user-images.githubusercontent.com/59128756/188512062-bc9d8d6c-af1d-46ce-9066-6d86a18307c3.png)

**Comment by svtdoug** [comment]
Thank you.  I had upgraded to 5.38 previously, but apparently it did not take.  Said I was still on 5.35.  I upgraded again just now and it seemed to take.  I'll watch it.  If it crashes again, I'll look for the additional .Net error.
